### PR TITLE
Test for unsupported features

### PR DESCRIPTION
**Pull request by MichaelClerx**
- Component units (#43)
- Reactions (was easy to do after the above)

### CODE REVIEW

File: cellmlmanip/parser.py [hunk 1]
```diff
@@ -171,14 +171,27 @@ def _add_components(self, model):
...
             # process the <variable> tags in this component
             variable_to_symbol = self._add_variables(element)
 
             # process the <math> tags in this component
             self._add_maths(element, variable_to_symbol)
 
+            # Raise error if component units are defined
+            component_units = element.findall(Parser.with_ns(XmlNs.CELLML, 'units'))
+            if component_units:
+                raise RuntimeError(
```
Comment (jonc125) [inline cellmlmanip/parser.py]: We seem to be using `ValueError` for other parser errors, so probably best to be consistent.

File: cellmlmanip/parser.py [hunk 2]
```diff
@@ -171,14 +171,27 @@ def _add_components(self, model):
...
+                    'Defining units inside components is not supported (found in component ' + name + ').')
+
+            # Raise error if reactions are defined
+            component_units = element.findall(Parser.with_ns(XmlNs.CELLML, 'reaction'))
```
Comment (jonc125) [inline cellmlmanip/parser.py]: I think you want to call the variable `component_reactions`!

File: tests/cellml_files/3.4.2.1.component_with_reactions.cellml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- CellML 1.0, 3.4.2.1: Components can contain reactions -->
+<!-- This model is from the CellML tests in https://github.com/MichaelClerx/cellml-validation
+     Used here to tests exceptions are raised if reactions are used. -->
```
Comment (jonc125) [inline tests/cellml_files/3.4.2.1.component_with_reactions.cellml]: s/tests/test/

**Comment by MichaelClerx** [comment]
Thanks!

**Pull request merged**